Dockerfile.in

```diff
@@ -1,5 +1,6 @@
-ARG base_image=registry.access.redhat.com/ubi9/ubi-minimal:9.4-1227.1726694542
-FROM ${base_image}
+ARG base_image=default
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4-1227.1726694542 AS default
+FROM ${base_image} as base
```

Comment:
Change looks good, just a couple of questions for my understanding.
1. Do we need the first `FROM` statement. How is that used?
2. Do we need `as base` in second `FROM` statement, also don't seem to be using the `base` name further.

Reply:
First `FROM` statement defines the default base image used and names it as `default`.
So if no `base_image` arg provided, the second `FROM` extended into `FROM default AS base`, which uses "local" reference to the image set in the first `FROM`
If `base_image` is provided (e.g. `my_image:latest`), the second `FROM` is extended to `FROM my_image:latest AS base`

`AS base` is not strictly necessary, it's there just for readability that this is a statement which sets the base image, but it can be removed if it would be clearer that way.

Reply:
thank you @hairyhum for explanation. makes sense.

Reply:
I would prefer removing the unnecessary `AS base` statement, but I will leave that to you.